docs/source/custom_sockets.rst

```diff
@@ -0,0 +1,244 @@
+.. _custom-sockets:
+
+=========================
+Custom Sockets in Syncano
+=========================
+
+``Syncano`` gives its users an ability to create Custom Sockets. What it means is that users can define 
+a very specific endpoints in their Syncano application, and use them exactly like they would any other Syncano 
+module (Classes, Scripts, etc), using standard API calls.
+Currently, Custom Sockets allow only one dependency - Scripts. It means that under the hood,
+each API call executes a Script, and result of this execution is returned as a result of the
+API call.
+
+Creating a custom socket
+------------------------
+
+To create a custom socket follow these steps::
+
+    import syncano
+    from syncano.models import CustomSocket, Endpoint, ScriptCall, ScriptDependency, RuntimeChoices
+    from syncano.connection import Connection
+
+    # 1. Initialize a custom socket.
+    custom_socket = CustomSocket(name='my_custom_socket')  # this will create an object in place (do API call)
+
+    # 2. Define endpoints.
+    my_endpoint = Endpoint(name='my_endpoint')  # no API call here
```

Comment:
FYI We will want to allow nested endpoints, e.g. Endpoint(name='my_endpoint/endpoint') which will then be called by /sockets/my_custom_socket/endpoints/my_endpoint/endpoint/ or /endpoints/sockets/my_endpoint/endpoint/.
Is it now in any way validated in library so that it wouldn't work once backend allows that? Couldn't find any validation, but just want to make sure